src/app/clusters/descriptor/descriptor.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Descriptor Server Cluster
+ ***************************************************************************/
+
+#include "af.h"
+#include <app/util/attribute-storage.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+
+#include "gen/af-structs.h"
+#include "gen/attribute-id.h"
+#include "gen/attribute-type.h"
+#include "gen/cluster-id.h"
+
+using namespace chip;
+
+constexpr const char * kErrorStr = "Descriptor cluster (0x%02x) Error setting '%s' attribute: 0x%02x";
+
+EmberAfStatus readWriteAttribute(uint8_t endpoint, AttributeId attributeId, bool write, uint8_t * buffer, int32_t index)
+{
+    EmberAfAttributeSearchRecord record;
+    record.endpoint         = endpoint;
+    record.clusterId        = ZCL_DESCRIPTOR_CLUSTER_ID;
+    record.clusterMask      = CLUSTER_MASK_SERVER;
+    record.manufacturerCode = EMBER_AF_NULL_MANUFACTURER_CODE;
+    record.attributeId      = attributeId;
+
+    return emAfReadOrWriteAttribute(&record, NULL, (uint8_t *) buffer, 0, write, index + 1);
```

Comment:
I have updated the method so it is only used for writing since there is no code reading attributes anymore. 

`write == true` or `readLength == 0` leads to the same underlying behavior: https://github.com/project-chip/connectedhomeip/blob/4f0ade5fcd50f99f53188ec401e18c2a22f47e14/src/app/util/attribute-storage.cpp#L377

Reply:
I opened #5897 for it.